### PR TITLE
added support to automagically add ucode to initrd 

### PR DESCRIPTION
**Pull request by advaithm**
# Pull Request Template

Make sure you've checked out the [contribution guideline](https://github.com/archlinux/archinstall/blob/master/CONTRIBUTING.md).<br>
Most of the guidelines are not enforced, but is heavily encouraged.

## Description
using the work I did in #217 and #212 I added support for arch install the required ucode packages and to add them to the initrd in systemd 
## Bugs and Issues

#177

## How Has This Been Tested?

I have not been able to test this on real hardware since I do not have a machine I can test this on. 
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code to the best of my abilities
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where possible/if applicable

### CODE REVIEW

File: archinstall/lib/installer.py [hunk 1]
```diff
@@ -364,13 +371,20 @@ def add_bootloader(self, bootloader='systemd-bootctl'):
...
 				entry.write(f'# Created by: archinstall\n')
 				entry.write(f'# Created on: {self.init_time}\n')
 				entry.write(f'title Arch Linux\n')
 				entry.write(f'linux /vmlinuz-linux\n')
+				if not isVM():
+					vendor = cpuVendor()
+					if vendor ==  "AuthenticAMD":
+						entry.write("initrd /amd-ucode.img")
+					elif vendor == "GenuineIntel":
+						entry.write("initrd /amd-ucode.img")
```
[the remark this answers is not in the document]
Comment (advaithm) [reply archinstall/lib/installer.py:385]: on genuine intel chips the vendor id is `GenuineIntel`. this a really old cpu I use as a server but its enough to demonstrate this.
```json
{
   "lscpu": [
      {"field":"Architecture:", "data":"x86_64"},
      {"field":"CPU op-mode(s):", "data":"32-bit, 64-bit"},
      {"field":"Byte Order:", "data":"Little Endian"},
      {"field":"Address sizes:", "data":"36 bits physical, 48 bits virtual"},
      {"field":"CPU(s):", "data":"2"},
      {"field":"On-line CPU(s) list:", "data":"0,1"},
      {"field":"Thread(s) per core:", "data":"1"},
      {"field":"Core(s) per socket:", "data":"2"},
      {"field":"Socket(s):", "data":"1"},
      {"field":"NUMA node(s):", "data":"1"},
      {"field":"Vendor ID:", "data":"GenuineIntel"},
      {"field":"CPU family:", "data":"6"},
      {"field":"Model:", "data":"23"},
      {"field":"Model name:", "data":"Intel(R) Core(TM)2 Duo CPU     E8400  @ 3.00GHz"},
      {"field":"Stepping:", "data":"10"},
      {"field":"CPU MHz:", "data":"3000.000"},
      {"field":"CPU max MHz:", "data":"3000.0000"},
      {"field":"CPU min MHz:", "data":"1998.0000"},
      {"field":"BogoMIPS:", "data":"5984.45"},
      {"field":"L1d cache:", "data":"64 KiB"},
      {"field":"L1i cache:", "data":"64 KiB"},
      {"field":"L2 cache:", "data":"6 MiB"},
      {"field":"NUMA node0 CPU(s):", "data":"0,1"},
      {"field":"Vulnerability Itlb multihit:", "data":"KVM: Mitigation: VMX unsupported"},
      {"field":"Vulnerability L1tf:", "data":"Mitigation; PTE Inversion"},
      {"field":"Vulnerability Mds:", "data":"Vulnerable: Clear CPU buffers attempted, no microcode; SMT disabled"},
      {"field":"Vulnerability Meltdown:", "data":"Mitigation; PTI"},
      {"field":"Vulnerability Spec store bypass:", "data":"Vulnerable"},
      {"field":"Vulnerability Spectre v1:", "data":"Mitigation; usercopy/swapgs barriers and __user pointer sanitization"},
      {"field":"Vulnerability Spectre v2:", "data":"Mitigation; Full generic retpoline, STIBP disabled, RSB filling"},
      {"field":"Vulnerability Srbds:", "data":"Not affected"},
      {"field":"Vulnerability Tsx async abort:", "data":"Not affected"},
      {"field":"Flags:", "data":"fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl cpuid aperfmperf pni dtes64 monitor ds_cpl smx est tm2 ssse3 cx16 xtpr pdcm sse4_1 xsave lahf_lm pti dtherm"}
   ]
}
```

**Comment by dylanmtaylor** [comment]
Error on line 385 - should be intel-ucode.

File: archinstall/lib/installer.py (hunk 1 above)
Comment (dylanmtaylor) [inline archinstall/lib/installer.py]: change to 						entry.write("initrd /intel-ucode.img")

**Comment by dylanmtaylor** [comment]
Looks good to me now.

**Comment by advaithm** [comment]
I'd like if some one could test this on real hardware

**Comment by Torxed** [comment]
Tested on a laptop and appears to be working (intel).
I'm not sure I tested it thoroughly, but there's plenty of time left for testing so I'm not to worried about it yet.

**Comment by dylanmtaylor** [comment]
> Tested on a laptop and appears to be working (intel).
> I'm not sure I tested it thoroughly, but there's plenty of time left for testing so I'm not to worried about it yet.

UEFI or MBR?

**Comment by Torxed** [comment]
> > Tested on a laptop and appears to be working (intel).
> > I'm not sure I tested it thoroughly, but there's plenty of time left for testing so I'm not to worried about it yet.
> 
> UEFI or MBR? I can test it too when I get a chance, I have a ThinkPad laying around I'm thinking about reinstalling anyways :)

I tested EFI. I still haven't gotten around to test the MBR yet ^^

**Comment by dylanmtaylor** [comment]
> > > Tested on a laptop and appears to be working (intel).
> > > I'm not sure I tested it thoroughly, but there's plenty of time left for testing so I'm not to worried about it yet.
> > 
> > 
> > UEFI or MBR? I can test it too when I get a chance, I have a ThinkPad laying around I'm thinking about reinstalling anyways :)
> 
> I tested EFI. I still haven't gotten around to test the MBR yet ^^

I'll probably test it with MBR once my changes land in 2.2.0 just to make sure things still work as expected (and I'm impatient so having the minimal install option REALLY helps me test faster)

**Pull request merged**